### PR TITLE
Using possible length info in ShortNumberInfo

### PR DESCRIPTION
**Pull request by lararennie**
Using new possible length information to calculate whether a short number is the right length or not. This made us realise that the data was missing for MY; this was added.

---
This change is [Reviewable](https://reviewable.io/reviews/googlei18n/libphonenumber/1407)

### CODE REVIEW

File: cpp/src/phonenumbers/shortnumberinfo.cc [hunk 1]
```diff
@@ -81,16 +81,19 @@ const PhoneMetadata* ShortNumberInfo::GetMetadataForRegion(
...
+// worth keeping this performance optimization.
 bool MatchesPossibleNumberAndNationalNumber(
-    const MatcherApi& matcher_api, const string& number,
-    const PhoneNumberDesc& number_desc) {
-  return matcher_api.MatchesPossibleNumber(number, number_desc) &&
-         matcher_api.MatchesNationalNumber(number, number_desc, false);
+    const MatcherApi& matcher_api,
+    const string& number,
+    const PhoneNumberDesc& desc) {
+  if (desc.possible_length_size() > 0 &&
```
Comment (keghani) [inline cpp/src/phonenumbers/shortnumberinfo.cc]: Flagging as possibly needing to change due to offline discussion

Comment (keghani) [inline cpp/src/phonenumbers/shortnumberinfo.cc]: Also below...

File: java/pending_code_changes.txt [hunk 2]
```diff
@@ -1 +1,5 @@
-
+Code changes:
+ - Using new possibleLengthInfo to decide whether a short number is the right
+length or not. This could result in more specific results; whereas before, a
```
Comment (keghani) [inline java/pending_code_changes.txt]: Please indent wrapped line to start at the same lace as "Using".

Comment (keghani) [inline java/pending_code_changes.txt]: What impact does this have on the versioning scheme? Could argue for x.0.0 or x.y.z - what did we do for regular numbers?

Comment (lararennie) [inline java/pending_code_changes.txt]: I would do x.y.z because it's like a metadata change: no one can or should change code because of it.

Comment (keghani) [inline java/pending_code_changes.txt]: Ack, thanks!

File: cpp/src/phonenumbers/shortnumberinfo.cc (hunk 1 above)
Comment (lararennie) [inline cpp/src/phonenumbers/shortnumberinfo.cc]: Changed all places (I think)

**Pull request merged**